### PR TITLE
Add useTxnActivity hook, reusable transaction table,  contract icons, and smart contract page 

### PR DESCRIPTION
**Pull request by BruceJi7**

### CODE REVIEW

File: src/helpers/index.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//     case "DRIVE":
+//       return "drive";
+
+//     case "ENGINE":
+//       return "engine";
+
+//     default:
+//       return "basic";
+//   }
+// }
```
Comment (BruceJi7) [inline src/helpers/index.ts:15]: The commented code here is for the production version, which only has the three contract types.
The uncommented code is for the test server which has a million contract types.

**Review by BruceJi7 — COMMENTED** [review]
I fixed some import order stuff I found, hope the rest is okay

File: src/hooks/useTxnActivity.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+type ParamProps = {
+  limit?: string | number;
+  chaincode?: string;
+  txId?: string | number;
+} | null;
+
+function useTxnActivity(params: ParamProps = null) {
+  const { channelHash, loadingChannelHash } = useChannelHash();
+
+  let axiosURL = `/txActivity/${channelHash}`;
```
Comment (ivanms1) [inline src/hooks/useTxnActivity.ts:18]: can this be moved outside the component?

File: src/pages/SmartContracts/ContractDetails/InfoTable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      </div>
+      <div className={styles.Cell}>
+        <div className={styles.CellTitle}>Type</div>
+        <div className={styles.CellContents}>
+          <ContractIcon contractName={smartContract.name} showLabel />
+        </div>
+      </div>
+      <div className={classnames(styles.Cell, styles.DateCell)}>
+        <div className={styles.CellTitle}>Updated</div>
+        {/* <div className={styles.CellContents}>{smartContract.updated}</div> */}
```
Comment (ivanms1) [inline src/pages/SmartContracts/ContractDetails/InfoTable/index.tsx:33]: remove please

File: src/pages/SmartContracts/index.tsx [hunk 4]
```diff
@@ -45,7 +36,8 @@ function SmartContract() {
       },
       {
         Header: "Last Updated",
-        accessor: "updated",
+        accessor: (row: any) => row.updated.toDateString(),
+        //todo - this should be time ago
```
Comment (ivanms1) [inline src/pages/SmartContracts/index.tsx:40]: you are waiting for junming's pr?

Comment (BruceJi7) [reply src/pages/SmartContracts/index.tsx:40]: Yes, hers has the time ago function

**Review by ivanms1 — APPROVED** [review]

**Pull request merged**